src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -414,8 +419,19 @@ private ProjectInfo CreateProjectInfoForDocument(string filePath)
                 metadataReferences: _metadataReferences);
 
             // Miscellaneous files projects are never fully loaded since, by definition, it won't know
-            // what the full set of information is.
-            return projectInfo.WithHasAllInformation(hasAllInformation: false);
+            // what the full set of information is except when the file is script code.
+            return projectInfo.WithHasAllInformation(hasAllInformation: sourceCodeKind == SourceCodeKind.Script);
+
+            SourceCodeKind GetSourceCodeKind()
```

Comment:
>GetSourceCodeKind [](start = 27, length = 17)

I don't think local function makes the code better. If you look at the function in isolation you don't know what the values of the captured locals are supposed to be. I'd either make it a static method or inline the code.

Reply:
sure